### PR TITLE
bpo-46045: Do not use POSIX semaphores on NetBSD

### PR DESCRIPTION
**Pull request by 0-wiz-0**
https://bugs.python.org/issue46045

### CODE REVIEW

File: Python/thread_pthread.h [hunk 1]
```diff
@@ -81,6 +81,8 @@
    we need to add 0 to make it work there as well. */
 #if (_POSIX_SEMAPHORES+0) == -1
 #define HAVE_BROKEN_POSIX_SEMAPHORES
+#elif defined(__NetBSD__)
+#define HAVE_BROKEN_POSIX_SEMAPHORES
```
Comment (tiran) [inline Python/thread_pthread.h]: ``configure.ac`` has a block to define ``HAVE_BROKEN_POSIX_SEMAPHORES`` for SunOS and AIX. Please move the NetBSD code there, too.

File: Misc/NEWS.d/next/Core and Builtins/2021-12-11-11-36-48.bpo-46045.sfThay.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Do not use POSIX semaphores on NetBSD
```
Comment (tiran) [inline Misc/NEWS.d/next/Core and Builtins/2021-12-11-11-36-48.bpo-46045.sfThay.rst:1]: The changelog also needs an explanation why POSIX semaphores cannot be used on NetBSD. Is there an issue with NetBSD kernel or do we need extra quirks to support NetBSD flavor of semaphores?

**Review by tiran — CHANGES_REQUESTED** [review]
Thank you for working on NetBSD support! Would you be interested to contribute a buildbot for NetBSD, too? As far as I know we don't have any CI for NetBSD.

File: Python/thread_pthread.h (hunk 1 above)
Comment (0-wiz-0) [reply Python/thread_pthread.h]: I've changed this as requested.
I noticed that the generated files from autoconf are in the repository, but I have a newer version installed, so I didn't include these files in the commit.

File: Misc/NEWS.d/next/Core and Builtins/2021-12-11-11-36-48.bpo-46045.sfThay.rst (hunk 2 above)
Comment (0-wiz-0) [reply Misc/NEWS.d/next/Core and Builtins/2021-12-11-11-36-48.bpo-46045.sfThay.rst:1]: Sorry, I don't know that.
I've tried using gdb to look at the python processes that are forked by the three mentioned tests, but there are a lot of them. Can you suggest a simple example where Python would use semaphores that I could use to file a NetBSD bug report?

**Comment by 0-wiz-0** [comment]
@tiran Could you point me to more information about running a build bot?
(Replies to the other comments see above.)

**Comment by 0-wiz-0** [comment]
Oh, and thanks for the review, @tiran!

**Comment by 0-wiz-0** [comment]
Can someone with autoconf-2.69 please regenerated the generated files?
@serhiy-storchaka ? @tarin ?

**Comment by 0-wiz-0** [comment]
Ok, autoconf was not hard. aclocal is from automake though and those files are generated by 1.16.3 (1.16.5 is out).
I had to manually stage the diff though because the generated configure script in the repository has `runstatedir` support, but autoconf-2.69 by default has not. Wondering about that one... but not a problem for this PR.

**Comment by 0-wiz-0** [comment]
I have made the requested changes; please review again

**Comment by tiran** [comment]
Can you run Linux containers on your system? We provider containers to regenerate autoconf files.

**Comment by 0-wiz-0** [comment]
No, I can't run Linux containers, but I've built a copy of autoconf 2.69 manually, so I can regenerate now.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
The change in `configure` matches generated by autoconf 2.69.

**Pull request merged**